File: app/src/main/java/com/example/studymanager/models/DTO/StudieVakHistoryDTO.kt
package com.example.studymanager.models.DTO

import com.example.studymanager.domain.StudieTask
import com.example.studymanager.domain.StudieVak
import com.example.studymanager.models.domain.StudieVakHistory

data class StudieVakHistoryDTO(
    var studieVakHistoryId: Int,
    var studieVakHistoryName: String,
    var aantalTasks: Int,
    var totaleStudieTijd: Long,
    var gebruikerId: Int

) {

    fun toModel(): StudieVakHistory {
        return StudieVakHistory(
            studieVakHistoryId = studieVakHistoryId,
            studieVakHistoryName = studieVakHistoryName,
            aantalTasks = aantalTasks,
            totaleStudieTijd = totaleStudieTijd
        )
    }
}

File: Backend/ProjectDashboard/DTO/StudieVakHistoryDTO.cs
using StudyAPI.Models.Domain;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace StudyAPI.DTO {
    public class StudieVakHistoryDTO {

        public int studieVakHistoryId { get; set; }
        [Required]
        public string studieVakHistoryName { get; set; }
        [Required]
        public int aantalTasks { get; set; }
        [Required]
        public long totaleStudieTijd { get; set; }
        [Required]
        public int gebruikerId { get; set; }
        public void updateFromModel(StudieVakHistory history) {
            history.StudieVakHistoryName = this.studieVakHistoryName;
            history.AantalTasks = this.aantalTasks;
            history.TotaleStudieTijd = this.totaleStudieTijd;
            history.GebruikerId = this.gebruikerId;
        }
    }
}
